### ISSUE TITLE
editor->propertyNames() chomps last character

### ISSUE THREAD

**Issue by vinsworldcom** [pryrt/Win32-Mechanize-NotepadPlusPlus#45]
Latest Win32::Mechanize::NotepadPlusPlus 0.003, hence forth referred to as "PerlScript".

Based on our discussion [here](https://community.notepad-plus-plus.org/topic/19486/help-with-json-and-scintilla-lexer/13), upon closer inspection, I noticed the editor->propertyNames() seems to be chomping the last character:

```
npp> print notepad->getLangType();
21
npp> print editor->propertyNames();
fold
fold.comment
fold.compact
fold.perl.pod
fold.perl.package
fold.perl.comment.explicit
fold.perl.at.els
```

But using NppExec, that last line should be "fold.perl.at.else" (last 'e' is missing above):

```
================ READY ================
SCI_SENDMSG SCI_PROPERTYNAMES 0 @""
================ READY ================
echo $(MSG_LPARAM)
fold
fold.comment
fold.compact
fold.perl.pod
fold.perl.package
fold.perl.comment.explicit
fold.perl.at.else
================ READY ================
```

It seems editor->propertyNames() returns a:

```
propertyNames
            editor->propertyNames()

        Retrieve a ΓÇÿ\nΓÇÖ separated list of properties understood by the
        current lexer.

        See Scintilla documentation for SCI_PROPERTYNAMES
        <https://www.scintilla.org/ScintillaDoc.html#SCI_PROPERTYNAMES>
```

(notice the weird characters - should be "backquote backslash n foward single quote") according to reading it in Notepad++ vs `perldoc` in PowerShell (as shown above) "list" which in Perl seems to really just be a variable with "\n" embedded?? In any case, thought maybe I was calling it wrong:

```
    for my $s ( split "\n", editor->propertyNames() ) {
...
```

But alas, the simple `print notepad->getLangType();` above shows the missing last character as well.

Cheers.

**Comment by pryrt** [comment]
Not surprising.  Some Scintilla messages return null-terminated strings, some are just based on length; my automatic wrappers sometimes mess up by stripping the last character assuming it's a NUL, or leaving an extra NUL at the end of the string.  I'll dig into it and see if I can get a fix.

(Incidentally, that's one of the reasons why I don't often post PerlScript code in the forums: every time I do, I seem to find a new bug to fix.  That, and you and I are probably the only forum readers who would be likely to have a perl installation already lying around _and_ want to automate Notepad++.  I guess if someone ever asks a question that strongly implies they do perl, I'll be more likely to post a PerlScript reply.)

**Comment by pryrt** [comment]
@vinsworldcom , hopefully will release fix (v0.003002) to CPAN in the next day or two.

**Comment by pryrt** [comment]
released to CPAN: https://metacpan.org/release/PETERCJ/Win32-Mechanize-NotepadPlusPlus-0.004
